src/gui/TimeLineWidget.cpp

```diff
@@ -451,15 +429,93 @@ void TimeLineWidget::mouseReleaseEvent( QMouseEvent* event )
 {
 	delete m_hint;
 	m_hint = NULL;
-	if (m_action == SelectSongTCO) { emit selectionFinished(); }
-	// Required when m_action == DragLoop, not harmful otherwise
-	if (m_loopPos[0] > m_loopPos[1]) { qSwap(m_loopPos[0], m_loopPos[1]); }
+
+	// Change action if mouse has not moved
+	if (m_action == Thresholded) { chooseMouseAction(event); }
+
+	mouseMoveEvent(event);
+
+	switch (m_action)
+	{
+		case SelectSongTCO:
+			emit selectionFinished();
+			break;
+
+		case ShowContextMenu:
+			contextMenuEvent(nullptr);
+			break;
+
+		default:
+			break;
+	}
+
 	m_action = NoAction;
 }
 
 
 
 
+void TimeLineWidget::chooseMouseAction(QMouseEvent* event)
+{
+	auto buttons = event->button() | event->buttons(); // include released button
+	auto mods = event->modifiers();
+	auto type = event->type();
+
+	// TODO: Read these from a config
+	// Unmodified LMB is reserved for playhead, unmodified RMB for context menu
+	// Shift or Ctrl modified press behavior is bound by the user
```

Comment:
Starting from this PR, the comment is not longer correct. Ctrl cannot (should not) be bound since it alone is the unquantize key. Oh, and in the next PR the MMB gets bound...